### PR TITLE
🎉 Source Shopify: updated `OAuth Spec`, allow `*.myshopify.com` shop names for `input configuration`

### PR DESCRIPTION
**Pull request by bazarnov**
## What
In order to complete OAuth Flow in the way the `Shopify` demands we need to allow the `*.myshopify.com` shop names pattern, add the `always_show: true` to the `shop` property for the field to be visible while making `OAuth` authentication for Cloud users, while OSS users remain intact.

## How
- removed the `"oauth_user_input_from_connector_config_specification"` property from `advanced_auth` spec
- added the `get_shop_name` method  to the `source.py` to extract the shop name from 2 input types (mainly from `*.myshopify.com` cases)
- covered with unit test
- tested locally + `dev-3`

## 🚨 User Impact 🚨
No impact is expected. This is not a braking change.

### CODE REVIEW

**Comment by bazarnov** [comment]
Successfully published the  `pre-released` version `[airbyte/source-shopify:0.5.1-dev.269733163f](https://hub.docker.com/r/airbyte/source-shopify/tags)` using this workflow: https://github.com/airbytehq/airbyte/actions/runs/5547045817

**Comment by bazarnov** [comment]
Successfully tested on `dev-3` using `connectors.versionOverrides` for `source-shopify` to `pre-released` tag  [airbyte/source-shopify:0.5.1-dev.269733163f](https://hub.docker.com/r/airbyte/source-shopify/tags) for workspace `c89869f2-7334-4fed-978e-fdd372f14dd9` rule: [Version switch for `source-shopify` to test the `OAuth Flow`](https://app.launchdarkly.com/default/dev-3/features/connectors.versionOverrides/targeting)

**Comment by octavia-squidington-iii** [comment]
##  source-shopify test report (commit [`269733163f`](https://github.com/airbytehq/airbyte/pull/28291/commits/269733163fe4fdddca74337f2d8b7ec440d24ac7)) - ❌

⏲️  Total pipeline duration: 30mn48s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-shopify/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-shopify docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Acceptance tests                                                      | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/baz_source-shopify-update-oauth-spec/1689336841/269733163fe4fdddca74337f2d8b7ec440d24ac7/source-shopify/0.5.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-shopify test
```

**Comment by octavia-squidington-iii** [comment]
##  source-shopify test report (commit [`2c0e9262ad`](https://github.com/airbytehq/airbyte/pull/28291/commits/2c0e9262adde97dbfa116b4f86f70455c9d19be6)) - ❌

⏲️  Total pipeline duration: 30mn38s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-shopify/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-shopify docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Acceptance tests                                                      | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/baz_source-shopify-update-oauth-spec/1689342647/2c0e9262adde97dbfa116b4f86f70455c9d19be6/source-shopify/0.5.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-shopify test
```

**Comment by octavia-squidington-iii** [comment]
##  source-shopify test report (commit [`ff1de8afcb`](https://github.com/airbytehq/airbyte/pull/28291/commits/ff1de8afcbe36c927cbfefe849b51186ba525f01)) - ❌

⏲️  Total pipeline duration: 30mn23s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-shopify/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-shopify docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Acceptance tests                                                      | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/baz_source-shopify-update-oauth-spec/1689344563/ff1de8afcbe36c927cbfefe849b51186ba525f01/source-shopify/0.5.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-shopify test
```

**Comment by octavia-squidington-iii** [comment]
##  source-shopify test report (commit [`ee2c3f2a14`](https://github.com/airbytehq/airbyte/pull/28291/commits/ee2c3f2a14559d4789dd404cb5f58d18eac5b093)) - ✅

⏲️  Total pipeline duration: 30mn32s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-shopify/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-shopify docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Acceptance tests                                                      | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/baz_source-shopify-update-oauth-spec/1689347809/ee2c3f2a14559d4789dd404cb5f58d18eac5b093/source-shopify/0.5.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-shopify test
```

**Comment by bazarnov** [comment]
Moving this to `On Hold` until we are approved for `Public` status by Shopify.

**Review by lazebnyi — APPROVED** [review]

File: airbyte-integrations/connectors/source-shopify/source_shopify/spec.json
Comment (brianjlai) [inline airbyte-integrations/connectors/source-shopify/source_shopify/spec.json:15]: If we're going to update the examples, we should also update the `description` to say:

"The name of your Shopify store found in the URL or the shop URL itself. For example, if your shop is located at https://NAME.myshopify.com, then shop can be set to either 'NAME' or 'NAME.myshopify.com'."

File: airbyte-integrations/connectors/source-shopify/source_shopify/spec.json
Comment (brianjlai) [inline airbyte-integrations/connectors/source-shopify/source_shopify/spec.json:131]: for my understanding why isn't think a breaking change? Is it because `oauth_user_input_from_connector_config_specification` isn't used anywhere in the code?

**Review by brianjlai — COMMENTED** [review]
changes make sense. Will approve once the merge conflicts are also addressed

File: airbyte-integrations/connectors/source-shopify/source_shopify/spec.json
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/spec.json:131]: The `breaking change` was considered before your review, as we decided to leave both options, there is no breaking change for now. Once everybody is successfully migrated to the `OAuth2.0`  we will manage the Cloud auth options with `spec handler` and that's it.

The `oauth_user_input_from_connector_config_specification` is part of the `advancedAuth` and not used anywhere in the code. As far as we automatically push the `shop` from the confirmation URL after the successful installation of the Airbyte<>Shopify public app, we no longer need to pass it manually for the OAuth Flow. Therefore it's removed.

File: airbyte-integrations/connectors/source-shopify/source_shopify/spec.json
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/spec.json:15]: Done. Make sense!

**Comment by octavia-squidington-iii** [comment]
##  source-shopify test report (commit [`03ccebbacd`](https://github.com/airbytehq/airbyte/pull/28291/commits/03ccebbacd8056b2acfc33fbcba7adae0fdb9f03)) - ✅

⏲️  Total pipeline duration: 24mn39s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-shopify/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-shopify docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Acceptance tests                                                      | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/baz_source-shopify-update-oauth-spec/1691501949/03ccebbacd8056b2acfc33fbcba7adae0fdb9f03/source-shopify/0.6.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:03ccebbacd8056b2acfc33fbcba7adae0fdb9f03)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-shopify test
```

File: airbyte-integrations/connectors/source-shopify/source_shopify/spec.json
Comment (brianjlai) [reply airbyte-integrations/connectors/source-shopify/source_shopify/spec.json:131]: sounds good, thanks for clarifying!

**Review by brianjlai — APPROVED** [review]
the changes look good to me!

**Pull request merged**
